src/Common/src/System/Security/Cryptography/CertificateValidation.Unix.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+using System.Globalization;
+using System.Net;
+using System.Net.Security;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    internal static class CertificateValidation
+    {
+        internal static SslPolicyErrors BuildChainAndVerifyProperties(X509Chain chain, X509Certificate2 remoteCertificate, bool checkCertName, string hostName)
```

Comment:
This is just refactored out of the existing CertificateValidationPal.Unix.cs file.